pkg/paths/path.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package paths
+
+import (
+	"strings"
+)
+
+//
+// To avoid confusion about when paths are encrypted, unencrypted, empty or
+// non existent, we create some wrapper types so that the compiler will complain
+// if someone attempts to use one in the wrong context.
+//
+
+// Unencrypted is an opaque type representing an unencrypted path.
+type Unencrypted struct {
```

Comment:
Shouldn't there also be a partially encrypted path type?

Reply:
No, a "path" refers to only the part after the bucket, and so is only encrypted or unencrypted.